### PR TITLE
Add torch.aten.flatten.using_ints and aten.MaxPool2d linalg lowering

### PR DESCRIPTION
**Pull request by cathyzhyi**
- torch.aten.flatten.using_ints to linalg lowering
- torch.aten.max_pool2d to linalg lowering
- Support torch.aten.conv2d for more flexible dilation and strides values

### CODE REVIEW

File: lib/Conversion/TorchToLinalg/TorchToLinalg.cpp [hunk 1]
```diff
@@ -839,6 +893,168 @@ struct ConvertElementwiseOp : ConversionPattern {
...
+                          .getResult(0);
+    Type newResultType = getTypeConverter()->convertType(op.getType());
+    rewriter.replaceOpWithNewOp<tensor::CastOp>(op, newResultType, maxPool2d);
+    return success();
+  }
+};
+} // namespace
+
+namespace {
+class ConvertAtenFlattenUsingIntsOp
```
Comment (silvasean) [inline lib/Conversion/TorchToLinalg/TorchToLinalg.cpp:1000]: this one is pretty tricky. can you add tests for it? (see tests for unsqueeze)

**Review by silvasean — CHANGES_REQUESTED** [review]

File: lib/Conversion/TorchToLinalg/TorchToLinalg.cpp (hunk 1 above)
Comment (cathyzhyi) [reply lib/Conversion/TorchToLinalg/TorchToLinalg.cpp:1000]: thanks for the suggestion. It does catch an error. Changed RefineTypes for flattening 0 rank case to set the resulting dim to be rank 1 with size 1.

Comment (silvasean) [reply lib/Conversion/TorchToLinalg/TorchToLinalg.cpp:1000]: Thanks. I was also referring to a .mlir file like test/Conversion/TorchToLinalg/unsqueeze.mlir which really helps isolate the lowering.

**Review by silvasean — CHANGES_REQUESTED** [review]

File: lib/Conversion/TorchToLinalg/TorchToLinalg.cpp (hunk 1 above)
Comment (cathyzhyi) [reply lib/Conversion/TorchToLinalg/TorchToLinalg.cpp:1000]: some how I missed those .mlir tests. Added suggested.

**Review by silvasean — APPROVED** [review]

**Pull request merged**